client/src/index.js

```diff
@@ -28,56 +39,97 @@ function persistReducer(state = { }, action) {
+      return action.comments
+    /*
+    case RELOAD_COMMENT:
+      return action
+
+    case PREVIEW_COMMENT:
+      return action
+    */
+    default:
+      return state
+  }
+}
+
 const reducer = combineReducers({
   persist: persistReducer,
+  comments: commentsReducer,
 })
 
 const store = createStore(reducer)
 
-const feedbackAppRoot = () => {
-  const feedbackAppRoot = document.createElement('div')
-  feedbackAppRoot.setAttribute('data-feedback-app-root', true)
-  document.body.appendChild(feedbackAppRoot)
-  return feedbackAppRoot
-}
-
 class MainView extends React.Component {
   constructor(props) {
     super(props)
 
-    this.handleClick = this.handleClick.bind(this)
+    this.handleSurveyPanelClick = this.handleSurveyPanelClick.bind(this)
+    this.toggleTagElementState = this.toggleTagElementState.bind(this)
+    this.handleElementTagged = this.handleElementTagged.bind(this)
 
     this.state = {
-      panelIsHidden: true,
-      buttonIsHidden: false,
+      surveyPanelIsHidden: true,
+      surveyButtonIsHidden: false,
+      taggingModeActive: false,
+      taggedElementXPath: '',
     }
   }
 
-  handleClick() {
+  handleSurveyPanelClick() {
     this.setState(state => ({
-      buttonIsHidden: !state.buttonIsHidden,
-      panelIsHidden: !state.panelIsHidden,
+      surveyPanelIsHidden: !state.surveyPanelIsHidden,
+      surveyButtonIsHidden: !state.surveyButtonIsHidden,
     }))
   }
 
+  toggleTagElementState() {
+    this.setState(state => ({
+      taggingModeActive: !state.taggingModeActive,
+    }))
+  }
+
+  handleElementTagged(event) {
+    const xPath = DomTagging.getCompleteElementXPath(event)
+    this.setState({
+      taggedElementXPath: xPath,
+    })
+    console.log('DOMT debug', 'index.js has tagged element', event, 'xpath:', xPath, 'state:', this.state.taggedElementXPath)
+  }
+
   render() {
-    const { buttonIsHidden, panelIsHidden } = this.state
+    const {
+      surveyButtonIsHidden,
+      surveyPanelIsHidden,
+      taggingModeActive,
+    } = this.state
 
     return (
-      <div>
-        <Button
-          hidden={buttonIsHidden}
-          onClick={this.handleClick}
+      <div
+        className={taggingModeActive ? css('feedback-app-container', 'tagging-mode-active') : css('feedback-app-container')}
+      >
+        <TagElementButton
+          active={taggingModeActive}
+          elementTagged={this.handleElementTagged}
+          toggleTagElementState={this.toggleTagElementState}
+        />
+        <OpenSurveyPanelButton
+          hidden={surveyButtonIsHidden}
+          onClick={this.handleSurveyPanelClick}
         />
-        <FloatingPanel
-          hidden={panelIsHidden}
-          onClick={this.handleClick}
+        <SurveyPanel
+          hidden={surveyPanelIsHidden}
+          onClick={this.handleSurveyPanelClick}
         />
+        <CommentPanel taggedElementXPath={this.state.taggedElementXPath} />
       </div>
     )
   }
 }
 
+// TODO: what is this
```

Comment:
@bqqbarbhg @joonalillfors

Reply:
In case there is multiple feedbacker forum instances, initialized only once

Reply:
FIXD